### ISSUE TITLE
refactor: simpler calculation of likelihoods

### ISSUE THREAD

**Issue by YodaEmbedding** [InterDigitalInc/CompressAI#182]
Regarding:

https://github.com/InterDigitalInc/CompressAI/blob/1259affe66da8a05750af8ae44d90989320c2103/compressai/entropy_models/entropy_models.py#L458-L469

The following:

```python
sign = -torch.sign(lower + upper)
sign = sign.detach()
likelihood = torch.abs(
    torch.sigmoid(sign * upper) - torch.sigmoid(sign * lower)
)
```

...seems to be equivalent to:

```python
likelihood = torch.abs(torch.sigmoid(upper) - torch.sigmoid(lower))
```

Furthermore, since `_logits_cumulative` is guaranteed to be a monotonically increasing scalar function on `R -> R`, we must have that `lower <= upper`. Thus, we can remove the abs altogether:

```python
likelihood = torch.sigmoid(upper) - torch.sigmoid(lower)
```

Unless I'm mistaken, there should be no difference in numerical stability either. 

---

"Proof" by experimentation:

```python
import torch

def likelihood_old(lower, upper):
    sign = -torch.sign(lower + upper)
    sign = sign.detach()
    likelihood = torch.abs(
        torch.sigmoid(sign * upper) - torch.sigmoid(sign * lower)
    )
    return likelihood

def likelihood_new(lower, upper):
    likelihood = torch.abs(torch.sigmoid(upper) - torch.sigmoid(lower))
    return likelihood

>>> n = 100
>>> lower, upper = (torch.randn(n) * 10**torch.randn(n), torch.randn(n) * 10**torch.randn(n))
>>> torch.isclose(f_old(lower, upper), f_new(lower, upper))
tensor([True, True, True, True, True, True, True, True, True, True, True, True,
        True, True, True, True, True, True, True, True, True, True, True, True,
        True, True, True, True, True, True, True, True, True, True, True, True,
        True, True, True, True, True, True, True, True, True, True, True, True,
        True, True, True, True, True, True, True, True, True, True, True, True,
        True, True, True, True, True, True, True, True, True, True, True, True,
        True, True, True, True, True, True, True, True, True, True, True, True,
        True, True, True, True, True, True, True, True, True, True, True, True,
        True, True, True, True])
```

---

NOTE: I haven't checked if the gradients are the same. I should probably do that...

**Comment by fracape** [comment]
Hi. 
Correct, the current implementation follows the original from Tensorflow compression (which changed since). No issues with back propagation with the sign? ( sign = sign.detach() )
